exchanges/kraken/kraken_websocket.go

```diff
@@ -222,16 +311,30 @@ func (k *Kraken) wsHandleData(respRaw []byte) error {
 						err,
 						respRaw)
 				}
-				if status.ErrorMessage != "" {
-					return fmt.Errorf("%s - err %s",
+
+				var isChannelExist bool
+				if status.RequestID > 0 {
+					isChannelExist = k.Websocket.Match.IncomingWithData(status.RequestID, respRaw)
+				}
+
+				if status.Status == "error" {
+					k.Websocket.DataHandler <- fmt.Errorf("%v Websocket status for RequestID %d: '%v'",
 						k.Name,
+						status.RequestID,
 						status.ErrorMessage)
+					return nil
```

Comment:
Same with here, just return the err you're sending to the datahandler instead

Reply:
done